taichi/backends/metal/device.cpp

```diff
@@ -224,11 +226,19 @@ class DeviceImpl : public Device {
   }
 
   DeviceAllocation allocate_memory(const AllocParams &params) override {
-    TI_NOT_IMPLEMENTED;
-    return DeviceAllocation{};
+    DeviceAllocation res;
+    res.device = this;
+    res.alloc_id = allocations_.size();
+
+    auto &ialloc = allocations_[res.alloc_id];  // "i" for internal
```

Comment:
super nit: I personally prefer to explicitly spell out the type when it's **short** to help code readability :P But I'm definitely aware that people have different opinions. Maybe good to align on this internally in the future? ;)

Reply:
+1 on explicitly writing out the type. Maybe https://google.github.io/styleguide/cppguide.html#Type_deduction ?